CONTRIBUTING.md

```diff
@@ -44,7 +44,14 @@ Before issuing a pull request we ask that you squash all your commits into a sin
 
 If you aren't sure how to squash multiple commits into one, Steve Klabnik wrote [a handy guide](http://blog.steveklabnik.com/posts/2012-11-08-how-to-squash-commits-in-a-github-pull-request) that you can refer to.
 
-We keep a [CHANGELOG](CHANGELOG.md) of all software changes with behavioural effects in ponyc. If your PR includes such changes (rather than say a documentation update), please make sure that as part of PR, you have also updated the CHANGELOG. The entries in the CHANGELOG are in time-ascending order, so please add your change description to the bottom of the appropriate section ("Fixed", "Added", or "Changed") of the "unreleased" log.
+We keep a [CHANGELOG](CHANGELOG.md) of all software changes with behavioural effects in ponyc. If your PR includes such changes (rather than say a documentation update), a Pony team member will do the following before merging it, so that the PR will be automatically added to the CHANGELOG:
+
+* Ensure that the ticket is tagged with one or more appropriate "changelog - *" labels - each label corresponds to a section of the changelog where this change will be automatically mentioned.
+* Ensure that the ticket title is appropriate - the title will be used as the summary of the change, so it should be appropriately formatted, including a ticket reference if the PR is a fix to an existing bug ticket.
+  * For example, an appropriate title for a PR that fixes a bug reported in issue ticket #98 might look like:
+  * *Fixed compiler crash related to tuple recovery (issue #98)*
+
+Once those conditions are met, the PR can be merged, and an automated system will immediately add the entry to the changelog. Keeping the changelog entries out of the file changes in the PR helps to avoid conflicts and other administrative headaches when many PRs are in progress.
```

Comment:
Having to merge PRs outside of the GitHub interface doesn't seem particularly attractive to me.  I'm trying to decrease the administrative and contributing burden, so that contributors can spend more time contributing and less time fussing with the changelog, and committers can spend more time triaging and merging and releasing, and less time fussing with the changelog.

My original scheme was to have a local script that was run once as part of every release, which would populate the changelog for the unreleased changes.  This was seen as undesirable by @SeanTAllen since it means we don't have a running "unreleased" section.

This new approach was created so that we can keep our running "unreleased" section, but still not have to fuss to much with the changelog.  Even after hearing your thoughts I don't see the extra commits as being a real problem or barrier to productivity in any significant way.  At the very least, it doesn't seem as important as the other concerns that pushed me into this approach.  But I suppose that reasonable people may differ here.